pkg/section/module.go

```diff
@@ -0,0 +1,78 @@
+package section
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/mod/modfile"
+
+	"github.com/daixiang0/gci/pkg/parse"
+	"github.com/daixiang0/gci/pkg/specificity"
+)
+
+const moduleType = "module"
+
+type Module struct {
+	ModulePaths []string
+}
+
+func (m *Module) MatchSpecificity(spec *parse.GciImports) specificity.MatchSpecificity {
+	for _, modPath := range m.ModulePaths {
+		if strings.HasPrefix(spec.Path, modPath) {
+			return specificity.Module{}
+		}
+	}
+
+	return specificity.MisMatch{}
+}
+
+func (m *Module) String() string {
+	return moduleType
+}
+
+func (m *Module) Type() string {
+	return moduleType
+}
+
+func (m *Module) Populate() error {
+	dir, err := os.Getwd()
+	if err != nil {
+		// TODO: warn
+		return nil
+	}
+	modFile := findModule(dir)
+	if modFile == "" {
+		return fmt.Errorf("local module lookup was requested, but could not find module file")
+	}
+
+	modContent, err := os.ReadFile(modFile)
+	if err != nil {
+		return fmt.Errorf("failed to read %s: %v", modFile, err)
+	}
+
+	modPath := modfile.ModulePath(modContent)
+	if modPath == "" {
+		return fmt.Errorf("could not read module path from %s", modFile)
+	}
+
+	m.ModulePaths = []string{modPath}
+	return nil
+}
+
+func findModule(dir string) string {
```

Comment:
Please check the golang community to see if there is a func for it.

Reply:
The canonical func used by the Go toolchain is `modload.findModuleRoot()`:

https://github.com/golang/go/blob/go1.21.3/src/cmd/go/internal/modload/init.go#L1434-L1452

and is used by `modload.Init()`, which handles `GOMODULES` et.al.:

https://github.com/golang/go/blob/go1.21.3/src/cmd/go/internal/modload/init.go#L370-L487

However, `modload` is an internal package.

Reply:
@dmke thanks.

@matthewhughes934 please try and comment where is the code from for better maintain.

Reply:
> @matthewhughes934 please try and comment where is the code from for better maintain.

I've gone with a new approach using https://pkg.go.dev/golang.org/x/tools/go/packages